cli/src/main/java/hudson/cli/CLI.java

```diff
@@ -103,16 +106,22 @@ public CLI(URL jenkins, ExecutorService exec) throws IOException, InterruptedExc
     }
 
     /**
-     * @deprecated 
-     *      Use {@link CLIConnectionFactory} to create {@link CLI}
+     * @deprecated Use {@link #CLI(hudson.cli.CLIConnectionFactory)
      */
     @Deprecated
     public CLI(URL jenkins, ExecutorService exec, String httpsProxyTunnel) throws IOException, InterruptedException {
         this(new CLIConnectionFactory().url(jenkins).executorService(exec).httpsProxyTunnel(httpsProxyTunnel));
     }
-    
-    /*package*/ CLI(CLIConnectionFactory factory) throws IOException, InterruptedException {
+
+    /**
+     * CLI connection to jenkins.
+     * @param factory builder with settings required for connection.
+     */
+    public CLI(CLIConnectionFactory factory) throws IOException, InterruptedException {
         URL jenkins = factory.jenkins;
+        if (jenkins == null) {
+            throw new IllegalArgumentException("Jenkins url must be specified in CLIConnectionFactory");
```

Comment:
`IllegalArgumentException` seems more appropriate to me, but should be thrown from `CLIConnectionFactory` IMO.